src/mantine-core/src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,203 @@
+  Item?: typeof CarouselItem;
+};
+
+const defaultProps: Partial<CarouselProps<any>> = {
+  prevButtonComponent: DefaultItem,
+  nextButtonComponent: DefaultItem,
+  indicatorButtonComponent: DefaultItem,
+  padding: 'md',
+  scrollOverflow: false,
+  height: 'md',
+  axis: 'x',
+  draggable: false,
+  slidesToScroll: 1,
+  withControls: true,
+  withIndicators: false,
+  shouldLoop: false,
+  startIndex: 0,
+};
+
+export const Carousel: CarouselCompoent = forwardRef(
+  <C extends React.ElementType = 'div'>(props: CarouselProps<C>, ref: PolymorphicRef<C>) => {
+    const {
+      prevButtonComponent: PrevButton,
+      nextButtonComponent: NextButton,
+      indicatorButtonComponent: IndicatorButton,
+      navigation,
+      axis,
+      component,
+      className,
+      padding,
+      radius,
+      draggable,
+      slidesToScroll,
+      scrollOverflow,
+      startIndex,
+      withIndicators,
+      shouldLoop,
+      height,
+      children,
+      classNames,
+      withControls,
+      styles,
+      ...others
+    } = useMantineDefaultProps('Carousel', defaultProps, props);
+    const { classes, cx } = useStyles(
+      { scrollOverflow, axis, height, padding, draggable },
+      { classNames, styles, name: 'Carousel' }
+    );
+    const _children = Children.toArray(children);
+
+    const content = filterChildrenByType(_children, CarouselItem).map((child, index) =>
+      React.cloneElement(child, {
+        padding,
+        key: index,
+      })
+    );
+
+    const [emblaRef, embla] = useEmblaCarousel({
+      skipSnaps: false,
+      axis,
+      draggable,
+      slidesToScroll,
+      startIndex,
+      loop: shouldLoop,
+    });
+    const [prevBtnEnabled, setPrevBtnEnabled] = useState(false);
+    const [nextBtnEnabled, setNextBtnEnabled] = useState(false);
+
+    const scrollPrev = useCallback(() => embla && embla.scrollPrev(), [embla]);
+    const scrollNext = useCallback(() => embla && embla.scrollNext(), [embla]);
+    const scrollTo = useCallback((index) => embla && embla.scrollTo(index), [embla]);
+    const [selectedIndex, setSelectedIndex] = useState(0);
+    const [scrollSnaps, setScrollSnaps] = useState([]);
+
+    const onSelect = useCallback(() => {
+      if (!embla) return;
+      setSelectedIndex(embla.selectedScrollSnap());
+      setPrevBtnEnabled(embla.canScrollPrev());
+      setNextBtnEnabled(embla.canScrollNext());
+    }, [embla, setSelectedIndex]);
+
+    useEffect(() => {
+      if (!embla) return;
+      onSelect();
+      setScrollSnaps(embla.scrollSnapList());
+      embla.on('select', onSelect);
+    }, [embla, setScrollSnaps, onSelect]);
+
+    return (
+      <Paper
+        className={cx(classes.root, className)}
+        radius={radius}
+        padding={padding}
+        ref={ref}
+        component={component as any}
+        {...others}
+      >
+        <div ref={emblaRef} className={classes.viewport}>
+          <Box<any> component="div" className={classes.container}>
```

Comment:
No need to set box type and component, should be just `<Box className={classes.container}>`, or maybe replace it with a div, since no Box props are being used.